lnd_test.go

```diff
@@ -2224,6 +2224,357 @@ func testMultiHopPayments(net *lntest.NetworkHarness, t *harnessTest) {
+	if err := net.ConnectNodes(ctxb, carol, net.Alice); err != nil {
+		t.Fatalf("unable to connect dave to alice: %v", err)
+	}
+	chanOpenUpdate, err := net.OpenChannel(ctxb, carol, net.Alice, chanAmt,
+		0, true)
+	if err != nil {
+		t.Fatalf("unable to open channel: %v", err)
+	}
+
+	// One block is enough to make the channel ready for use, since the
+	// nodes have defaultNumConfs=1 set.
+	block := mineBlocks(t, net, 1)[0]
+	chanPointPrivate, err := net.WaitForChannelOpen(ctxb, chanOpenUpdate)
+	if err != nil {
+		t.Fatalf("error while waiting for channel open: %v", err)
+	}
+	fundingTxID, err := chainhash.NewHash(chanPointPrivate.FundingTxid)
+	if err != nil {
+		t.Fatalf("unable to create sha hash: %v", err)
+	}
+	assertTxInBlock(t, block, fundingTxID)
+
+	// The channel should be listed in the peer information returned by
+	// both peers.
+	privateFundPoint := wire.OutPoint{
+		Hash:  *fundingTxID,
+		Index: chanPointPrivate.OutputIndex,
+	}
+	if err := net.AssertChannelExists(ctxb, carol, &privateFundPoint); err != nil {
+		t.Fatalf("unable to assert channel existence: %v", err)
+	}
+	if err := net.AssertChannelExists(ctxb, net.Alice, &privateFundPoint); err != nil {
+		t.Fatalf("unable to assert channel existence: %v", err)
+	}
+
+	// The channel should be available for payments between Carol and Alice.
+	// We check this by sending payments from Carol to Bob, that collectively
+	// would deplete at least one of Carol's channels.
+
+	// Create 2 invoices for Bob, each of 70k satoshis. Since each of
+	// Carol's channels is of size 100k, these payments cannot succeed
+	// by only using one of the channels.
+	const numPayments = 2
+	const paymentAmt = 70000
+	payReqs := make([]string, numPayments)
+	for i := 0; i < numPayments; i++ {
+		invoice := &lnrpc.Invoice{
+			Memo:  "testing",
+			Value: paymentAmt,
+		}
+		resp, err := net.Bob.AddInvoice(ctxb, invoice)
+		if err != nil {
+			t.Fatalf("unable to add invoice: %v", err)
+		}
+
+		payReqs[i] = resp.PaymentRequest
+	}
+
+	time.Sleep(time.Millisecond * 50)
+
+	// Let Carol pay the invoices.
+	ctxt, _ = context.WithTimeout(ctxb, timeout)
+	err = completePaymentRequests(ctxt, carol, payReqs, true)
+	if err != nil {
+		t.Fatalf("unable to send payments: %v", err)
+	}
+
+	// When asserting the amount of satoshis moved, we'll factor in the
+	// default base fee, as we didn't modify the fee structure when
+	// creating the seed nodes in the network.
+	const baseFee = 1
+
+	assertAmountPaid := func(node *lntest.HarnessNode, chanPoint wire.OutPoint,
+		amountSent, amountReceived int64) {
+
+		channelName := ""
+		switch chanPoint {
+		case aliceFundPoint:
+			channelName = "Alice(local) => Bob(remote)"
+		case daveFundPoint:
+			channelName = "Dave(local) => Alice(remote)"
+		case carolFundPoint:
+			channelName = "Carol(local) => Dave(remote)"
+		case privateFundPoint:
+			channelName = "Carol(local) [private=>] Alice(remote)"
+		}
+
+		checkAmountPaid := func() error {
```

Comment:
This looks to be cop-pasta'd from elsewhere in the integration tests. Instead, we should just extract this function to be made publicly callable.

Reply:
You caught me 🙊 

.. done